src/loan.rs

```diff
@@ -482,9 +482,9 @@ pub struct Borrower0 {
 
 impl Borrower0 {
     #[allow(clippy::too_many_arguments)]
-    pub async fn new<R, CS, CF>(
+    pub async fn new<R>(
         rng: &mut R,
-        coin_selector: CS,
```

Comment:
Unless the API changes are truly minimal (like a rename), we should almost always favor deprecation over breaking it. Just to make our own lives easier because you can work in smaller, incremental chunks.

Reply:
We spent some time thinking about deprecating this, but we decided against it because:

1. We thought it was actually a minimal change. Users of the library only need to move the coin selection up a couple of lines and use arguments they're already passing in to this constructor.
2. We are handling the breaking change for our only user ourselves with https://github.com/comit-network/waves/pull/202.
3. Creating another constructor with a name other than `new` for a struct that is almost certainly going to disappear in the next few weeks seems overkill.

I think we want to prioritise deprecating over introducing breaking changes (so your comment is very welcome), but I think this one is fine.